lib/reservation.rb

```diff
@@ -0,0 +1,113 @@
+require 'awesome_print'
+require 'date'
+require 'csv'
+require_relative 'room'
+
+module Hotel
+  class Reservation
+    attr_reader :reservation_id, :check_in, :check_out, :number_of_days
+    attr_accessor :status, :room_number, :room
+    VALID_STATUS = [:pending, :denied, :confirmed]
+
+    def initialize(reservation_id, check_in, number_of_days, status = :pending, room = nil, room_number = nil)
+      @reservation_id = reservation_id
+      
+      if check_in.class != Date
+        raise ArgumentError.new("Invaid, check-in date must be Date.")
+      end
+      @check_in = check_in
+
+
+      if number_of_days <= 0
+        raise ArgumentError.new("Invalid, number of days must be greater than 0")
```

Comment:
It's helpful to include the invalid arguments in the message when raising an `ArgumentError`:

```suggestion
        raise ArgumentError.new("Invalid, number of days must be greater than 0.  Got: #{number_of_days}")
```